eng/common/TestResources/SubConfig-Helpers.ps1

```diff
@@ -19,7 +19,15 @@ function GetUserName() {
 function GetBaseName([string]$user, [string]$serviceDirectoryName) {
     # Handle service directories in nested directories, e.g. `data/aztables`
     $serviceDirectorySafeName = $serviceDirectoryName -replace '[\./\\]', ''
-    return "$user$serviceDirectorySafeName".ToLowerInvariant()
+    $BaseName =  "$user$serviceDirectorySafeName".ToLowerInvariant()
+    $BaseName = "loadtestingSDK"
+    # Key-Vault accounts have a limit of 24 characters
+    if ($BaseName.Length -gt 24) {
```

Comment:
It would be better here to throw.  Truncating may cause name collisions and result in failures that are less clear.

Reply:
@jsquire it does make sense truncating can cause the issues listed, however, I wanted to understand that if we throw it here, then how would users like me be doing it? because when we run tests this script is also invoked and it might be a problem. 

How about we ask for input for BaseName as a fallback instead of truncating?

Reply:
Fair point, I forgot about that scenario.  Probably better that we validate the parameter upfront in the test resources script, when provided, and fallback here such that if we exceeded the limit, we truncate.